doc/source/develop/testing-formulas.rst

```diff
@@ -0,0 +1,264 @@
+
+Kitchen spin an instances in Docker, Vagrant, OpenStack environment, etc. based on configured driver.
+Instance is configured as salt minion, where the configuration is defined by ``.kitchen.yml`` and ``tests/pillar/*.sls``
+
+Override your specific needs with ``.kitchen.<backend|local>.yml`` that you may load as:
+``KITCHEN_LOCAL_YAML=.kitchen.<driver>.yml kitchen <action> <suite>``.
+
+Example: ``KITCHEN_LOCAL_YAML=.kitchen.local kitchen verify server-ubuntu-1404 -t tests/integration``.
+
+Test Kitchen then allows you execute several action to perform your testing under configured conditions:
+
+1. *create*, provision an test instance (VM, container)
+2. *converge*, run a provisioner (shell script, kitchen-salt)
+3. *verify*, run a verification (inspec, other may be added)
+4. *destroy*
+
+
+
+Verifying deployment
+^^^^^^^^^^^^^^^^^^^^
+
+There is couple of verifier plugins that are shipped with Test Kitchen. They allow to run simple bash scripts and checking
+it's exit codes to run specific purpose based frameworks.
+
+The `Busser <https://github.com/test-kitchen/busser>`_ *Verifier* goes with test-kitchen by default.
+It is used to setup and run tests implemented in ``<repo>/test/integration``. It guess and installs the particular driver to tested instance.
+By default `InSpec <https://github.com/chef/kitchen-inspec>`_ is expected.
+
+You may avoid to install busser framework if you configure specific verifier in ``.kitchen.yml`` and install it kitchen plugin locally:
+
+	verifier:
+		name: inspec
+
+If you would to write another verification scripts than InSpec store them in ``<repo>/tests/integration/<suite>/<busser>/*``.
+``Busser <https://github.com/test-kitchen/busser>`` is a test setup and execution framework under test kitchen.
+
+
+Implement integration tests under ``<repo>/tests/integration/<suite>/<busser>/*`` directory with ``_spec.rb`` filename
+suffix.
+
+**InSpec**
+
+InSpec is native validation framework for Test Kitchen and as such don't require usage of ``<busser>`` folder. Thus the tests may by
+stored directly under ``<repo>/tests/integration/<suite>/*``
+
+Additional resources.
+
+* https://github.com/chef/inspec
+* https://github.com/chef/kitchen-inspec
+
+
+Example verification scripts under `tests/integration` folder of the formula:
+
+.. code-block:: shell
+
+  tests
+  ├── integration
+  │   ├── default
+  │   │   └── default_testcase_spec.rb  # Written in InSpec
+  │   ├── backupmx
+  │   │   └── serverspec                # <Verifier framework>
+  │   │       └── backupmx_spec.rb      # Written in ServerSpec
+  │   ├── helpers
+  │   │   └── serverspec
+  │   │       └── spec_helper.rb
+  │   ├── relay
+  │   │   └── serverspec
+  │   │       └── relay_spec.rb
+  │   └── server
+  │       └── serverspec
+  │           ├── aliases_spec.rb
+  │           └── server_spec.rb
+  ├── pillar
+  │   ├── backupmx.sls
+  │   ├── relay.sls
+  │   └── server.sls
+  └── run_tests.sh
+
+
+Requirements
+^^^^^^^^^^^^^^^^^^
```

Comment:
Too long underline. Run make html to see and fix warnings/error and final document.